### PR TITLE
Move some usages of match to matches!

### PR DESCRIPTION
**Pull request by kjeremy**

### CODE REVIEW

File: crates/ra_parser/src/syntax_kind/generated.rs [hunk 1]
```diff
@@ -251,33 +251,122 @@ pub enum SyntaxKind {
...
-        match self {
-            AS_KW | ASYNC_KW | AWAIT_KW | BOX_KW | BREAK_KW | CONST_KW | CONTINUE_KW | CRATE_KW
-            | DYN_KW | ELSE_KW | ENUM_KW | EXTERN_KW | FALSE_KW | FN_KW | FOR_KW | IF_KW
-            | IMPL_KW | IN_KW | LET_KW | LOOP_KW | MACRO_KW | MATCH_KW | MOD_KW | MOVE_KW
-            | MUT_KW | PUB_KW | REF_KW | RETURN_KW | SELF_KW | STATIC_KW | STRUCT_KW | SUPER_KW
-            | TRAIT_KW | TRUE_KW | TRY_KW | TYPE_KW | UNSAFE_KW | USE_KW | WHERE_KW | WHILE_KW
-            | AUTO_KW | DEFAULT_KW | EXISTENTIAL_KW | UNION_KW => true,
-            _ => false,
-        }
+        matches!(
```
Comment (matklad) [inline crates/ra_parser/src/syntax_kind/generated.rs]: I think for longer matches like these, `matches!` is a strictly worse

File: crates/rust-analyzer/src/cli.rs [hunk 2]
```diff
@@ -22,16 +22,10 @@ pub enum Verbosity {
...
-            _ => false,
-        }
+        matches!(self, Verbosity::Verbose | Verbosity::Spammy)
     }
     pub fn is_spammy(self) -> bool {
-        match self {
-            Verbosity::Spammy => true,
-            _ => false,
-        }
+        matches!(self, Verbosity::Spammy)
```
Comment (matklad) [inline crates/rust-analyzer/src/cli.rs]: This one is short, but we also don't use it as a part of a bigger expression (like `if`), so an explicit match reads better to me.

File: crates/ra_syntax/src/ast/generated.rs [hunk 3]
```diff
@@ -50,10 +47,7 @@ impl std::fmt::Display for FnDef {
 }
 impl AstNode for FnDef {
     fn can_cast(kind: SyntaxKind) -> bool {
-        match kind {
-            FN_DEF => true,
-            _ => false,
-        }
+        matches!(kind, FN_DEF)
```
Comment (matklad) [inline crates/ra_syntax/src/ast/generated.rs]: I was thinking at one point about special-casing this with just `kind == FN_DEF`.

**Comment by matklad** [comment]
Yeah, not a fan, it seems like in all cases `matches` is worse than either an explicit match, or an `==`.

**Comment by kjeremy** [comment]
Ok!

**Pull request closed without merging**